### DISCUSSION TITLE
Maven private repository

### DISCUSSION THREAD

**Discussion by remi-cogito** [oss-review-toolkit/ort#9455] [General]
How to set up docker image to authenticate with private maven repository?

**Comment by sschuberth** [comment]
Please see https://oss-review-toolkit.org/ort/docs/guides/docker#authenticating-with-a-private-git-repository-fails which describes the solution based on a `.netrc` file. You basically need the same, but with a Maven `settings.xml` file mounted to `~/.m2/settings.xml` inside the container.

**Comment by BinchenQ** [comment]
I have tried with  `/home/ort/.m2/settings.xml` and `~/.m2/settings.xml`and `/root/.m2/settings.xml`, it did not work.
```
status code: 401, reason phrase:  (401) and 'parent.relativePath' points at no local POM @ line 19, column 13
```

I could run `mvn clean package ` successfully in this container, but ort analyze failed. Any dependency in private jfrog could not be downloaded.

**Comment by arunsmartarrow** [comment]
I am also facing the same issue as described by @BinchenQ.

@sschuberth: Can you please advise here, thanks

**Comment by sschuberth** [comment]
> status code: 401, reason phrase:  (401) and 'parent.relativePath' points at no local POM @ line 19, column 13

In this specific case from @BinchenQ it seems like the parent POM is hosted elsewhere, like another host, which might require different authentication than the primary project under analysis. Hard to say without more details...

**Comment by MNesche** [comment]
Hi, in my case, the following appplied:

Running ORT locally in a VM with Ubuntu:
Only the solution with .gitconfig worked for private repositories.
The .netrc file is recognized when running with --debug, but the request leads always to "Not Authorized".

Running ORT locally in Windows:
Only the solution with .netrc worked for private repositories, the same .gitconfig who worked on the VM, didn't.
The log with --debug showed, that ORT tried to find a .netrc file, which was not present, so that one led to "Not Authorized".
